Install/update sh script: Avoid overwriting existing gdb config

mAddmHacks opened the pull request:
## Description

<!-- Describe technically what your patch does. -->
As a GDB user I need to keep my existing config when installing GEF

<!-- Why is this change required? What problem does it solve? -->
The `~/.gdbinit` content is currently completely removed

<!-- Why is this patch will make a better world? -->
Still happy to keep my GDB config while it's now powered by GEF!

<!-- How does this look? Add a screenshot if you can -->
Replaced move with copy for the backup
[Upgrade] Check if GEF is already sourced, and replace version
[Install] Add GEF script at 1st line

<!-- Annotate your PR with label proper labels (architecture impacted, type of improvement,
etc.) ->

## Checklist

<!-- N.B.: Your patch won't be reviewed unless fulfilling the following base requirements: -->
<!--- Put an `x` in all the boxes that are complete, or that don't apply -->
-  [x] My code follows the code style of this project.
-  [x] My change includes a change to the documentation, if required.
-  [x] If my change adds new code, [adequate tests](docs/testing.md) have been added.
-  [x] I have read and agree to the **CONTRIBUTING** document.

hugsy commented:
LGTMerge when tests pass

Grazfather commented on scripts/gef.sh:
```diff
@@ -10,14 +10,16 @@ fi
 if [ -f "${HOME}/.gef-${tag}.py" ]; then
-    echo "source ~/.gef-${tag}.py" > ~/.gdbinit
+    grep -q "source ~/.gef-" ~/.gdbinit && \
+    sed -i "s#source ~/.gef-.*\.py#source ~/.gef-${tag}.py#" ~/.gdbinit || \
```
let's indent so it's clearer it's part of the previous.
Also, can you comment what this is doing?
Looks like it finds source of any gef and updates the tag?

mAddmHacks replied:
Expanded if statements for better visibility
Comments added

mAddmHacks commented on scripts/gef.sh:
```diff
@@ -10,14 +10,16 @@ fi
-    echo "source ~/.gef-${tag}.py" > ~/.gdbinit
+    grep -q "source ~/.gef-" ~/.gdbinit && \
+    sed -i "s#source ~/.gef-.*\.py#source ~/.gef-${tag}.py#" ~/.gdbinit || \
+    sed -i "1i source ~/.gef-${tag}.py" ~/.gdbinit
```
Comment added

mAddmHacks commented on scripts/gef.sh:
```diff
@@ -10,14 +10,16 @@ fi
 
 if [ -f "${HOME}/.gef-${tag}.py" ]; then
-    echo "source ~/.gef-${tag}.py" > ~/.gdbinit
+    grep -q "source ~/.gef-" ~/.gdbinit && \
```
Not sure if you want to support this. The updater script writes GEF in a specific place

mAddmHacks replied:
Marking as resolved, if we want this, it should go to another PR IMO :)

mAddmHacks commented:
> @mAddmHacks
> 
> The code looks good but tests are currently failing. Once they pass we can merge your PR

I added a missing behavior if gdbinit didn't exist, thanks to your test, now fixed

hugsy reviewed (changes requested):
Just a tiny suggestion, then LGTM
